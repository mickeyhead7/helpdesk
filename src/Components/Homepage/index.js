import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BodyHeader from '@Components/BodyHeader';
import ArticleList from '@Components/ArticleList';
import ConnectedTagList from '@Components/TagList';
import I18nProvider from '@packages/Components/i18n/Provider';
import Translate from '@packages/Components/i18n/Translate';
import enGB from './languages/en-GB';

const phrases = {
  'en-GB': enGB,
};

export const Homepage = ({ articles, locale, tags }) => (
  <I18nProvider locale={locale} phrases={phrases}>
    <BodyHeader>
      <h1>
        <Translate locale={locale} phrase="messages.welcome" />
      </h1>
    </BodyHeader>
    <section>
      <header>
        <h2>
          <Translate locale={locale} phrase="messages.latestArticles" />
        </h2>
      </header>
      <ArticleList articles={articles} />
    </section>
    <section>
      <ConnectedTagList tags={tags} />
    </section>
  </I18nProvider>
);

Homepage.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    node: PropTypes.shape({
      frontmatter: PropTypes.shape({
        excerpt: PropTypes.string.isRequired,
        path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  })).isRequired,
  locale: PropTypes.string.isRequired,
  tags: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
};

const mapStateToProps = ({ i18n }) => ({
  locale: i18n.locale,
});

export default connect(mapStateToProps)(Homepage);
