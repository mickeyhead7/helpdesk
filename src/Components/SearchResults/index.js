import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BodyHeader from '@Components/BodyHeader';
import ArticleList from '@Components/ArticleList';
import I18nProvider from '@packages/Components/i18n/Provider';
import Translate from '@packages/Components/i18n/Translate';
import enGB from './languages/en-GB';

const phrases = {
  'en-GB': enGB,
};

export const SearchResults = ({
  articles,
  locale,
  search,
}) => (
  <I18nProvider locale={locale} phrases={phrases}>
    <BodyHeader>
      <h1>
        <Translate locale={locale} phrase="messages.searchResultsFor" params={{ search }} />
      </h1>
    </BodyHeader>
    <ArticleList articles={articles} />
  </I18nProvider>
);

SearchResults.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    node: PropTypes.shape({
      frontmatter: PropTypes.shape({
        excerpt: PropTypes.string.isRequired,
        path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  })).isRequired,
  locale: PropTypes.string.isRequired,
  search: PropTypes.string.isRequired,
};

const mapStateToProps = ({ i18n }) => ({
  locale: i18n.locale,
});

export default connect(mapStateToProps)(SearchResults);
