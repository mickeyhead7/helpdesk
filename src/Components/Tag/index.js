import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BodyHeader from '@Components/BodyHeader';
import Pagination from '@packages/patterns/Pagination';
import ArticleList from '@Components/ArticleList';
import I18nProvider from '@packages/Components/i18n/Provider';
import Translate from '@packages/Components/i18n/Translate';
import enGB from './languages/en-GB';

const phrases = {
  'en-GB': enGB,
};

export const Tag = ({
  locale,
  articles,
  index,
  first,
  last,
  pageCount,
  tagName,
}) => (
  <I18nProvider locale={locale} phrases={phrases}>
    <BodyHeader>
      <h1>
        <Translate locale={locale} phrase="messages.tagArticles" params={{ tagName }} />
      </h1>
    </BodyHeader>
    <ArticleList articles={articles} />
    <Pagination
      index={index}
      first={first}
      last={last}
      pageCount={pageCount}
      urlPrefix={`/tags/${tagName}`}
    />
  </I18nProvider>
);

const mapStateToProps = ({ i18n }) => ({
  locale: i18n.locale,
});

Tag.propTypes = {
  locale: PropTypes.string.isRequired,
  articles: PropTypes.arrayOf(PropTypes.shape({
    node: PropTypes.shape({
      frontmatter: PropTypes.shape({
        path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }).isRequired,
    }),
  })),
  index: PropTypes.number.isRequired,
  first: PropTypes.bool.isRequired,
  last: PropTypes.bool.isRequired,
  pageCount: PropTypes.number.isRequired,
  tagName: PropTypes.string.isRequired,
};

Tag.defaultProps = {
  articles: [],
};

export default connect(mapStateToProps)(Tag);
