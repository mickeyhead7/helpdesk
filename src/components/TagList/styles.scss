@import '../../themes/default/settings.media.scss';

.tagsList {
    .tagsHeader {
        .allTagsLink {
            display: inline-block;
            margin: 0 0 1em;
        }

        @media #{$desktopUp} {
            display: flex;
            align-items: center;

            .allTagsLink {
                margin: 0 0 1em auto;
            }
        }
    }

    .tagLinks {
        $tabletUp: "only screen and (min-width: 40.0625em)";
        $desktopUp: "only screen and (min-width: 60.0625em)";

        font-family: sans-serif;
        margin: 0 -1em 2em;

        @media #{$tabletUp} {
            display: flex;
            flex-wrap: wrap;
        }

        .tag {
            list-style: none;
            margin: 0 1em 1em;
            
            @media #{$tabletUp} {
                display: flex;
                flex-direction: column;
                width: calc(50% - 2em);
            }

            @media #{$desktopUp} {
                width: calc(25% - 2em);
            }
        }
    }
}
