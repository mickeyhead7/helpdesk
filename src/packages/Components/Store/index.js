import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import store from '@store';

const Store = ({ children }) => (
  <Provider store={store}>
    {children}
  </Provider>
);

Store.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Store;
