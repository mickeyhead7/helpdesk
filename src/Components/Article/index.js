import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Link from 'gatsby-link';
import BodyHeader from '@Components/BodyHeader';
import Card from '@packages/patterns/Card';
import I18nProvider from '@packages/Components/i18n/Provider';
import Translate from '@packages/Components/i18n/Translate';
import enGB from './languages/en-GB';

import './styles.scss';

const phrases = {
  'en-GB': enGB,
};

export const Article = ({
  locale,
  frontmatter,
  html,
  prev,
  next,
}) => {
  const { date, title } = frontmatter;

  return (
    <I18nProvider locale={locale} phrases={phrases}>
      <article className="articleFull">
        <BodyHeader>
          <div className="articleHeader">
            <h1>{title}</h1>
            <small className="articleDate">
              <Translate locale={locale} phrase="messages.postedDate" params={{ date }} />
            </small>
          </div>
        </BodyHeader>
        <section
          className="articleContent"
          dangerouslySetInnerHTML={{ __html: html }}
        />
        <section>
          {(prev || next) && (
            <Fragment>
              <h2>
                <Translate locale={locale} phrase="messages.moreArticles" />
              </h2>
              <ul className="articleLinks">
                {prev && (
                  <li>
                    <Card>
                      <h3>
                        <Translate locale={locale} phrase="messages.previousArticle" />
                      </h3>
                      <h4>
                        <Link to={prev.frontmatter.path}>
                          {prev.frontmatter.title}
                        </Link>
                      </h4>
                      <p>
                        {prev.frontmatter.excerpt}
                      </p>
                    </Card>
                  </li>
                )}
                {next && (
                  <li>
                    <Card>
                      <h3>
                        <Translate locale={locale} phrase="messages.nextArticle" />
                      </h3>
                      <h4>
                        <Link to={next.frontmatter.path}>
                          {next.frontmatter.title}
                        </Link>
                      </h4>
                      <p>
                        {next.frontmatter.excerpt}
                      </p>
                    </Card>
                  </li>
                )}
              </ul>
            </Fragment>
          )}
        </section>
      </article>
    </I18nProvider>
  );
};

Article.propTypes = {
  locale: PropTypes.string.isRequired,
  frontmatter: PropTypes.shape({
    date: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
  html: PropTypes.node.isRequired,
  next: PropTypes.shape({
    frontmatter: PropTypes.shape({
      path: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }).isRequired,
  }),
  prev: PropTypes.shape({
    frontmatter: PropTypes.shape({
      path: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }).isRequired,
  }),
};

Article.defaultProps = {
  next: null,
  prev: null,
};

const mapStateToProps = ({ i18n }) => ({
  locale: i18n.locale,
});

export default connect(mapStateToProps)(Article);
