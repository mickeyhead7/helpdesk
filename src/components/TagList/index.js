import React from 'react';
import Link from 'gatsby-link';
import propTypes from 'prop-types';

import './styles.scss';

const TagList = ({ tags }) => (
  <section className="tagsList">
    <header className="tagsHeader">
      <h2>Popular tags</h2>
      <Link className="allTagsLink" to="/tags">
        View all tags
      </Link>
    </header>
    <ul className="tagLinks">
      {tags.map((tag, index) =>
        <li className="tag" key={index}>
          <Link to={`/tags/${tag}`}>{tag}</Link>
        </li>
      )}
    </ul>
  </section>
);

TagList.propTypes = {
  tags: propTypes.array.isRequired,
};

export default TagList;
