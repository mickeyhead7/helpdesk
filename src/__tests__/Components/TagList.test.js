import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Link from 'gatsby-link';
import { TagList } from '../../Components/TagList';

const tags = ['tag-one', 'tag-two', 'tag-three'];

test('Header', () => {
  const props = {
    locale: 'en-GB',
    tags,
  };
  const component = shallow(<TagList {...props} />);
  const expectedTagsLength = tags.length + 1;

  expect(component.find(Link)).toHaveLength(expectedTagsLength);
  expect(toJson(component)).toMatchSnapshot();
});
