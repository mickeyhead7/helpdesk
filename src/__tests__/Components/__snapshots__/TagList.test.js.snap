// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<Connect(Provider)
  locale="en-GB"
  phrases={
    Object {
      "en-GB": Object {
        "links": Object {
          "allTags": "View all tags",
        },
        "messages": Object {
          "popularTags": "Popular tags",
        },
      },
    }
  }
>
  <section
    className="tagsList"
  >
    <header
      className="tagsHeader"
    >
      <h2>
        <Connect(Translate)
          locale="en-GB"
          phrase="messages.popularTags"
        />
      </h2>
      <Link
        className="allTagsLink"
        to="/tags"
      >
        <Connect(Translate)
          locale="en-GB"
          phrase="links.allTags"
        />
      </Link>
    </header>
    <ul
      className="tagLinks"
    >
      <li
        className="tag"
        key="tag-one"
      >
        <Link
          to="/tags/tag-one"
        >
          tag-one
        </Link>
      </li>
      <li
        className="tag"
        key="tag-two"
      >
        <Link
          to="/tags/tag-two"
        >
          tag-two
        </Link>
      </li>
      <li
        className="tag"
        key="tag-three"
      >
        <Link
          to="/tags/tag-three"
        >
          tag-three
        </Link>
      </li>
    </ul>
  </section>
</Connect(Provider)>
`;
